server/text_generation_server/models/custom_modeling/mpt_modeling.py

```diff
@@ -616,15 +650,13 @@ def forward(
 
 
 def _cast_if_autocast_enabled(tensor):
-    if torch.is_autocast_enabled():
-        if tensor.device.type == "cuda":
-            dtype = torch.get_autocast_gpu_dtype()
-        elif tensor.device.type == "cpu":
-            dtype = torch.get_autocast_cpu_dtype()
-        else:
-            raise NotImplementedError()
-        return tensor.to(dtype=dtype)
-    return tensor
+    if tensor.device.type == "cuda":
+        dtype = torch.get_autocast_gpu_dtype()
+    elif tensor.device.type == "cpu":
+        dtype = torch.get_autocast_cpu_dtype()
+    else:
+        raise NotImplementedError()
+    return tensor.to(dtype=dtype)
```

Comment:
Why do we need to modify this ? Autocast causes performance issues more than it helps sometimes, and it's hard to debug given the indirect nature of it.

Can you provide more background for the necessity here ?

Reply:
There was issue earlier while making changes to `mpt_modelling.py` where we need to force the autocast, and so the "if" check was removed.
After re-testing with the current `mpt_modelling.py` with the "if" restored, it does work now. So, the "if" should be retained to avoid any possible performance issues.

Reply:
Thnaks a lot.